### DISCUSSION TITLE
How to prefer ipv4 binding when sending email ?

### DISCUSSION THREAD

**Discussion by dtrckd** [postalserver/postal#2274] [Help with using Postal]
I am experiencing spam blacklisting issues from some domains due to ipv6 range allocation and would want to send email with the ipv4 address instead of ipv6.

Is there a way to ask Postal to prefer ipv4 address binding when sending mail ?

Note: Just to mention, there is a smtp_address_preference setting in Postfix to address that.

**Comment by willpower232** [comment]
if the receiving email server has IPv6 available then Postal will always default to using that if its bound to `::` so has access to your IPv6 addresses

You could try binding to `0.0.0.0` instead which should stop you using IPv6 altogether but if you're having spam issues then it might not just be down to your IP address.

**Reply by dtrckd** [reply]
But in his case, I wouldn't be able to receive email through ipv6, which might cause lost of emails sent from domains that would only support ipv6 ?

**Reply by willpower232** [reply]
In which case, I don't think Postal can do what you need to in this case. Are you able to request yourself removed from the blocklists?

**Reply by dtrckd** [reply]
Yes, I asked for it (it's spamhaus.org). But, now they redirected me to the ISP asking a /64 ipv6 range suited for email service, as it seems there is an IP that is not mine but in the same ipv6 range that has bad reputation...

I don't know, /64 range email seems to be standard, I don't get the policy of spamhaus.org here, if someone in the /64 range have a bad reputation, every IP in the range also have ??

**Reply by willpower232** [reply]
Interesting, I guess with such a vast quantity of addresses available, they're presuming people are allocated specific blocks rather than individual ones maybe? Unfortunate but I guess it depends what support you can get from your server/internet company.

**Reply by dtrckd** [reply]
Yes, I think it's like that too.

I am just getting back that to this point, but I am really curious about it :
1.  do you think that implementing a feature supporting `smtp_address_preference = ipv4||ipv6` ", which will try to send email binding to one address/interface first will be that hard ?
2. if it is proposed as a contribution, does the postal team will be interested to merge it eventually ?

**Reply by willpower232** [reply]
It doesn't sound impossible but its way out of what I know about how to bind to ports. If you or someone makes a PR at least then the code will exist and people that need it can make their own docker image if it doesn't get merged.

**Reply by dtrckd** [reply]
I resume this discussion because I am still  curious why this issue does is not get more attention (Are people using Postal more for internal emailing than marketig ?) ;
More specifically, when getting a softfail  with an error message from the remote server that suggest that the email was rejected because of the sender IP in a spam list, one could try to send with the other IP addresses (if they are multiples addresses, but basically IPv4 and ipv6 cohabitation seems to be ubiquitous.)  which could solve the error if the other address is not listed...

**Reply by dtrckd** [reply]
Some interesting point to consider already mentioned here https://github.com/postalserver/postal/issues/605

**Comment by vulnfind** [comment]
where can I find the setting of smtp_address_preference ???? is it in postal.yml and if yes under which heading i have to put it under??? smtp: or smtp_server: ???

**Reply by stef157** [reply]
Hi, is there some ETA about this smtp_address_preference ?
